eclipse-collections-code-generator/src/main/resources/impl/bag/mutable/primitiveHashBag.stg

```diff
@@ -745,6 +745,28 @@ public class <name>HashBag
         return array;
     }
 
+    @Override
+    public <type>[] toArray(<type>[] array)
+    {
+        if (array.length \< this.size())
+        {
+            array = new <type>[this.size()];
+        }
+
+        int[] index = {0};
+
+        <type>[] finalBypass = array;
+        this.forEachWithOccurrences((<type> each, int occurrences) ->
+        {
+            for (int i = 0; i \< occurrences; i++)
+            {
+                finalBypass[index[0]] = each;
+                index[0]++;
+            }
+        });
```

Comment:
Nitpick: I don't see much advantage of using forEachWithOccurrences when the output is of size `size()`. Might as well delegate to forEachWithIndex or forEach?